akka-actor/src/main/scala/akka/actor/ActorSystem.scala

```diff
@@ -1179,29 +1179,25 @@ private[akka] class ActorSystemImpl(
   private def loadExtensions(): Unit = {
 
     /*
-     * @param throwOnLoadFail Throw exception when an extension fails to load (needed for backwards compatibility)
+     * @param throwOnLoadFail
+     *  Throw exception when an extension fails to load (needed for backwards compatibility.
+     *    when the extension cannot be found at all we throw regardless of this setting)
```

Comment:
That's not true, it's the opposite as far as I understand. If class not found (dynamicAccess) throws we honor the `throwOnLoadFail` flag, but if the extension itself throws (lookup) we don't.

That was what I found inconsistent, but maybe we could still throw the original exception instead of wrapping in RuntimeException for the `lookup` case.

Reply:
I'd say that the test is wrong. akka.extensions and akka.library-extensions are treated differently so test should maybe use akka.library-extensions instead

Reply:
Agreed that comment was backwards.

The tests seem consistent with the behavior though:
* fail the actor system if an extension listed in akka.extensions fails to start
* log an error if an extension listed in akka.extensions cannot be loaded
* fail the actor system if a library-extension fails to start
* fail the actor system if a library-extension cannot be loaded

Which do you think is wrong?

Reply:
I was thinking this is wrong:

> fail the actor system if an extension listed in akka.extensions fails to start

Why would that be different from load extension?  akka.extensions should log and continue.

But I could have missed something.

Reply:
From 14c08df7 I suspect that it went like this:
* First we had only akka.extensions, and logged on failure
* Then we added akka.library-extensions, and figured it would be better to throw on failure - but felt we couldn't just start failing in a patch release, so made kept the old behavior for akka.extensions

Perhaps we should schedule throwing instead of logging for 2.7.0? :)

Reply:
Thanks for looking into it. Let's leave it as is. It's not important.